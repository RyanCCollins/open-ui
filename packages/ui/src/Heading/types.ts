import * as React from 'react';
import { Margin } from '../Paragraph/types';
import Heading from './index';
export type TextAligment = 'center' | 'left' | 'right' | 'justify';
export type Tag = 'h1' | 'h2' | 'h3' | 'h4' | 'h5';

export type Margin = 'none' | 'small' | 'medium' | 'large';

export interface MarginSizeMap {
  none: number;
  small: number;
  medium: number;
  large: number;
}

export interface HeadingProps extends React.Props<Heading> {
  color?: string;
  textAlign?: string;
  tag?: Tag;
  truncate?: boolean;
  upcase?: boolean;
  margin?: Margin;
}
