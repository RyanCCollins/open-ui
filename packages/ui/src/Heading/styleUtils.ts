import { Tag } from './types';
import remStringFromPX from '../utils';

const sizeMap = {
  h1: 36,
  h2: 30,
  h3: 24,
  h4: 18,
  h5: 16,
};

export const calculateSize = (tag: Tag): string => remStringFromPX(sizeMap[tag]);
