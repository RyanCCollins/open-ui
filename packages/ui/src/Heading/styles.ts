import { marginCss } from '../Paragraph/styles';
import { calculateSize } from './styleUtils';
const styled = require('styled-components').default;
const { css } = require('styled-components');

const truncateCss = (truncate: boolean) => {
  if (truncate) {
    return css`
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    `;
  }
  return '';
};

const textTransformCss = (upcase: boolean) => {
  if (upcase) {
    return css`
      text-transform: uppercase;
    `;
  }
  return '';
};

export const H1 = styled.h1`
  font-size: ${(props) => calculateSize(props.tag)};
  text-align: ${(props) => props.textAlign};
  color: ${(props) => props.color};
  ${(props) => truncateCss(props.truncate)};
  ${(props) => textTransformCss(props.upcase)};
  ${(props) => marginCss(props.margin)};
`;

export const H2 = styled.h2`
  font-size: ${(props) => calculateSize(props.tag)};
  text-align: ${(props) => props.textAlign};
  color: ${(props) => props.color};
  ${(props) => truncateCss(props.truncate)};
  ${(props) => textTransformCss(props.upcase)};
  ${(props) => marginCss(props.margin)};
`;

export const H3 = styled.h3`
  font-size: ${(props) => calculateSize(props.tag)};
  text-align: ${(props) => props.textAlign};
  color: ${(props) => props.color};
  ${(props) => truncateCss(props.truncate)};
  ${(props) => textTransformCss(props.upcase)};
  ${(props) => marginCss(props.margin)};
`;

export const H4 = styled.h4`
  font-size: ${(props) => calculateSize(props.tag)};
  text-align: ${(props) => props.textAlign};
  color: ${(props) => props.color};
  ${(props) => truncateCss(props.truncate)};
  ${(props) => textTransformCss(props.upcase)};
  ${(props) => marginCss(props.margin)};
`;

export const H5 = styled.h5`
  font-size: ${(props) => calculateSize(props.tag)};
  text-align: ${(props) => props.textAlign};
  color: ${(props) => props.color};
  ${(props) => truncateCss(props.truncate)};
  ${(props) => textTransformCss(props.upcase)};
  ${(props) => marginCss(props.margin)};
`;
