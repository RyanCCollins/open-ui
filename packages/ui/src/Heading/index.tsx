import * as React from 'react';
import { HeadingProps } from './types';
import { H1, H2, H3, H4, H5 } from './styles';

class Heading extends React.Component<HeadingProps, undefined> {
  public static defaultProps: HeadingProps = {
    color: '#007acc',
    textAlign: 'center',
    tag: 'h1',
    truncate: false,
    upcase: false,
    margin: 'medium',
  };
  public render() {
    const { children, tag, ...rest } = this.props;
    switch (tag) {
    case 'h2':
      return (
      <H2 tag={tag} {...rest}>
        {children}
      </H2>
      );
    case 'h3':
      return (
      <H3 tag={tag} {...rest}>
        {children}
      </H3>
      );
    case 'h4':
      return (
      <H4 tag={tag} {...rest}>
        {children}
      </H4>
      );
    case 'h5':
      return (
      <H5 tag={tag} {...rest}>
        {children}
      </H5>
      );
    default:
      return (
        <H1 tag={tag} {...rest}>
          {children}
        </H1>
      );
    }
  }
}

export default Heading;
