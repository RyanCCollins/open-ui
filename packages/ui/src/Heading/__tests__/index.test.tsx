import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Heading from '../';
import { h2Props, h3Props, h4Props, h5Props } from './__mocks__/headingProps.mock';

describe('<Heading />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Heading>
        Test
      </Heading>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with h2 props', () => {
    const wrapper = shallow(
      <Heading {...h2Props}>
        Test
      </Heading>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with h3 props', () => {
    const wrapper = shallow(
      <Heading {...h3Props}>
        Test
      </Heading>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with h4 props', () => {
    const wrapper = shallow(
      <Heading {...h4Props}>
        Test
      </Heading>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with h5 props', () => {
    const wrapper = shallow(
      <Heading {...h5Props}>
        Test
      </Heading>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
