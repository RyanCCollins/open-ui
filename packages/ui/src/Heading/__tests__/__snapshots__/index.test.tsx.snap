// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render with default props 1`] = `
<styled.h1
  color="#007acc"
  margin="medium"
  tag="h1"
  textAlign="center"
  truncate={false}
  upcase={false}
>
  Test
</styled.h1>
`;

exports[`<Heading /> should render with h2 props 1`] = `
<styled.h2
  color="#333"
  margin="none"
  tag="h2"
  textAlign="left"
  truncate={true}
  upcase={false}
>
  Test
</styled.h2>
`;

exports[`<Heading /> should render with h3 props 1`] = `
<styled.h3
  color="#333"
  margin="none"
  tag="h3"
  textAlign="left"
  truncate={true}
  upcase={false}
>
  Test
</styled.h3>
`;

exports[`<Heading /> should render with h4 props 1`] = `
<styled.h4
  color="#333"
  margin="none"
  tag="h4"
  textAlign="left"
  truncate={true}
  upcase={false}
>
  Test
</styled.h4>
`;

exports[`<Heading /> should render with h5 props 1`] = `
<styled.h5
  color="#333"
  margin="none"
  tag="h5"
  textAlign="left"
  truncate={true}
  upcase={false}
>
  Test
</styled.h5>
`;
